Sources/PackageSets/API.swift

```diff
@@ -0,0 +1,179 @@
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke when result becomes available
+    func listPackageSets(
+        identifers: Set<PackageSetsModel.PackageSetIdentifier>?,
+        in profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<[PackageSetsModel.PackageSet], Error>) -> Void
+    )
+
+    /// Refreshes all configured package sets.
+    ///
+    /// - Parameters:
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke after triggering a refresh for the configured package sets.
+    func refreshPackageSets(
+        in profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<[PackageSetsModel.PackageSetSource], Error>) -> Void
+    )
+
+    /// Adds a package set.
+    ///
+    /// - Parameters:
+    ///   - source: The package set's source
+    ///   - order: Optional. The order that the `PackageSet` should take after being added to the list.
+    ///            By default the new group is appended to the end (i.e., the least relevant order).
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke with the updated `PackageSet`s
+    func addPackageSet(
+        _ source: PackageSetsModel.PackageSetSource,
+        order: Int?,
+        to profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<PackageSetsModel.PackageSet, Error>) -> Void
+    )
+
+    /// Removes a package set.
+    ///
+    /// - Parameters:
+    ///   - source: The package set's source
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke with the updated `PackageSet`s
+    func removePackageSet(
+        _ source: PackageSetsModel.PackageSetSource,
+        from profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<Void, Error>) -> Void
+    )
+
+    /// Moves a package set to a different order.
+    ///
+    /// - Parameters:
+    ///   - id: The identifier of the `PackageSet` to be moved
+    ///   - order: The new order that the `PackageSet` should be positioned after the move
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke with the updated `PackageSet`s
+    func movePackageSet(
+        _ source: PackageSetsModel.PackageSetSource,
+        to order: Int,
+        in profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<Void, Error>) -> Void
+    )
+
+    /// Returns information about a package set. The set is not required to be in the configured list. If
+    /// not found locally, the group will be fetched from the source.
```

Comment:
nit: should this say "set" instead of "group"?

Reply:
renamed this to collections and addressed